splice_machine/src/test/java/com/splicemachine/foreignkeys/ForeignKeyActionIT.java

```diff
@@ -0,0 +1,339 @@
+            s.executeUpdate("insert into P values(1,10),(2,20),(3,30),(4,40)");
+
+            s.executeUpdate("create table C1 (a int, b int, CONSTRAINT FK_1 FOREIGN KEY (a) REFERENCES P(a))");
+            s.executeUpdate("insert into C1 values(1,10),(1,15),(2,20),(2,20),(3,30),(3,35)");
+
+            assertQueryFail(s,"delete from P where a = 2","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+            assertQueryFail(s,"update P set a=-1 where a = 2","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+
+            s.executeUpdate("create table C2 (a int, b int, CONSTRAINT FK_2 FOREIGN KEY (b) REFERENCES P(b))");
+            s.executeUpdate("insert into C2 values(4,40)");
+
+            // verify NEW FK constraint works
+            assertQueryFail(s,"delete from P where a = 4","Operation on table 'P' caused a violation of foreign key constraint 'FK_2' for key (B).  The statement has been rolled back.");
+            assertQueryFail(s,"update P set b=-1 where a = 4","Operation on table 'P' caused a violation of foreign key constraint 'FK_2' for key (B).  The statement has been rolled back.");
+
+            // verify FIRST FK constraint STILL works
+            assertQueryFail(s,"delete from P where a = 1","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+            assertQueryFail(s,"update P set a=-1 where a = 1","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+        }
+    }
+
+    @Test
+    public void onDeleteNoActionPrimaryKeySuccessAfterDeleteReference() throws Exception {
+        try(Statement s = conn.createStatement()){
+            createDatabaseObjects1(s);
+
+            assertQueryFail(s,"delete from P where a = 2","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+            assertQueryFail(s,"update P set a=-1 where a = 2","Operation on table 'P' caused a violation of foreign key constraint 'FK_1' for key (A).  The statement has been rolled back.");
+
+            // delete references
+            s.executeUpdate("delete from C where a=2");
+
+            // now delete from parent should succeed
+            assertEquals(4L,StatementUtils.onlyLong(s,"select count(*) from C"));
+            assertEquals(1L,s.executeUpdate("delete from P where a = 2"));
+        }
+    }
+
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    //
+    // FK - ON DELETE SET NULL
+    //
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
```

Comment:
What will happen if the foreign key column is not nullable? Do we have such a test case?

`create table SRT (a int primary key, b int not null, constraint fk foreign key (B) references SRT(a) on delete set null);`

Probably should throw already in binding phase since it's semantically conflicting.

Reply:
You are correct, we get a semantic check error:
```
create table SRT (a int primary key, b int not null, constraint fk foreign key (B) references SRT(a) on delete set null);
ERROR 42834: SET NULL cannot be specified because FOREIGN KEY 'FK'  cannot contain null values.
```